### ISSUE TITLE
Numpy in windows 10 when has overflow returns  wrong number without error

### ISSUE THREAD

**Issue by reza1615** [numpy/numpy#13369]
In Windows 10, when Numpy has overflow it shows a wrong number without error.

In Linux it doesn't have this problem.

```python
import numpy as np
a= [4, 7, 6, 5, 4, 5, 6, 8, 2, 8, 4, 8, 9]
print (np.prod(a))
>> -579076096
```
The correct one is
```python
import numpy as np
a= [4, 7, 6, 5, 4, 5, 6, 8, 2, 8, 4, 8, 9]
print (np.prod(a, dtype='int64'))
>> 23514624000
```
Numpy should show some errors of overflow
[Related issue](https://stackoverflow.com/questions/55767586/numpy-array-prod-vs-multiply-list-items)

**Comment by charris** [comment]
This is probably due to C `long` on windows being 32 bits, so you need to specify `int64`. NumPy ignores overflow, it follows the C standard of modular arithmetic.

**Comment by eric-wieser** [comment]
I think the underlying cause here would be fixed by #12332

**Comment by seberg** [comment]
I agree, this is basically a duplicate of gh-12332 and issues related to reporting integer overflows in general. So closing in favor of those.
